plugins/workflow/category/src/Extension/Category.php

```diff
@@ -0,0 +1,259 @@
+     *
+     * @param   Form      $form  The form
+     * @param   \stdClass    $data  The data
+     *
+     * @return  void
+     *
+     * @since   DEPLOY_VERSION
+     */
+    protected function disableCategoryField(Form $form, $data)
+    {
+        // Get the current category ID value
+        $catid = $form->getValue('catid');
+
+        $form->setFieldAttribute('catid', 'readonly', 'true');
+        $form->setFieldAttribute('catid', 'value', $catid);
+
+        $js = "
+            document.addEventListener('DOMContentLoaded', function() {
+                var categoryField = document.getElementById('jform_catid');
+                if (categoryField) {
+                    categoryField.disabled = true;
+                }
+            });
+        ";
+        $this->getApplication()->getDocument()->addScriptDeclaration($js);
+    }
+
+
+    /**
+     * Method to handle the workflow transition event.
+     *
+     * @param   WorkflowTransitionEvent  $event  The event object
+     *
+     * @return  void
+     *
+     * @since   DEPLOY_VERSION
+     */
+    public function onWorkflowAfterTransition(WorkflowTransitionEvent $event): void
+    {
+        $app = $this->getApplication();
+        $pks = $event->getArgument('pks');
+        $transition = $event->getArgument('transition');
+
+        if (!is_object($transition) || !($transition->options instanceof Registry) || empty($transition->options)) {
+            $app->enqueueMessage('PLG_WORKFLOW_CATEGORY_INVALID_TRANSITION');
+            return;
+        }
+
+        $options = $transition->options;
+        $categoryId = (int) $options->get('category_id');
+
+        if (empty($pks) || !is_array($pks)) {
+            $app->enqueueMessage(Text::_('PLG_WORKFLOW_CATEGORY_NO_PRIMARY_KEY'), 'error');
+            return;
+        }
+
+        $errors = 0;
+
+        foreach ($pks as $pk) {
+            if (!self::processArticle($pk, $categoryId)) {
+                $errors++;
+            }
+        }
+
+        if ($errors > 0) {
+            $app->enqueueMessage(Text::plural('PLG_WORKFLOW_CATEGORY_ERRORS', $errors), 'warning');
+        }
+    }
+
+    /**
+     * Process the article and update its category.
+     *
+     * @param   int                  $pk         The primary key
+     * @param   int                  $categoryId The category ID
+     *
+     * @return  bool
+     *
+     * @since   DEPLOY_VERSION
+     */
+    private function processArticle($pk, $categoryId): bool
+    {
+        $app = $this->getApplication();
+        $result = false;
+
+        try {
+
+            $component = $this->getApplication()->bootComponent('com_content');
+            $modelName = $component->getModelName('com_workflow.article');
+
+            $articleTable = $component->getMVCFactory()->createModel($modelName, $this->getApplication()->getName(), ['ignore_request' => true])
+                ->getTable();
+            if (!$articleTable->load($pk)) {
```

Comment:
For coding an "else" in an "if-statement" is most of the time not best practise. A general rule of thumb is: no "else" but working with [early returns](https://dev.to/jpswade/return-early-12o5).